File: tests/functional/syntax_features/background_feature/steps.rb
begin require 'rspec/expectations'; rescue LoadError; require 'spec/expectations'; end

count = 0

Given /^that world count is (\d+)$/ do |arg1|
  count.should == 0
end

When /^I add one to the world$/ do
  count = 1
end

Then /^world count should be (\d+)$/ do |arg1|
  true
end

File: tests/functional/syntax_features/background_feature/steps.py
from lettuce import step, world
from nose.tools import assert_equals


@step("Given that world count is 5")
def set(step):
    world.count = 5
    assert_equals(world.count, 5)

@step("Then world count should be 6")
def check(step):
    assert_equals(world.count, 6)

@step("Then world count should be 5")
def check(step):
    assert_equals(world.count,5)

@step("When I add one to the world")
def check(step):
    world.count += 1

